### PR TITLE
wifi: esp32: update `link_mode` to reflect actual negotiated PHY mode

### PR DESCRIPTION
**Pull request by ndrs-pst**
Updated `link_mode` assignment to reflect the actual negotiated PHY mode by using `esp_wifi_sta_get_negotiated_phymode`.

Since `ap_info.phy_11b` ... `ap_info.phy_11ax` represent the access point capabilities and not the actual current link mode.

### CODE REVIEW

**Review by uLipe — DISMISSED** [review]

File: drivers/wifi/esp32/src/esp_wifi_drv.c [hunk 1]
```diff
@@ -619,14 +619,21 @@ static int esp32_wifi_status(const struct device *dev, struct wifi_iface_status
...
-			if (ap_info.phy_11b) {
-				status->link_mode = WIFI_1;
-			} else if (ap_info.phy_11g) {
-				status->link_mode = WIFI_3;
-			} else if (ap_info.phy_11n) {
-				status->link_mode = WIFI_4;
-			} else if (ap_info.phy_11ax) {
-				status->link_mode = WIFI_6;
+			wifi_phy_mode_t phy_mode;
+			esp_err_t err;
```
Comment (jukkar) [inline drivers/wifi/esp32/src/esp_wifi_drv.c:623]: Please introduce variables at the start of the block

Comment (ndrs-pst) [reply drivers/wifi/esp32/src/esp_wifi_drv.c]: Got it.

**Review by jukkar — APPROVED** [review]

**Review by sylvioalves — APPROVED** [review]

**Review by marekmatej — APPROVED** [review]

**Comment by ndrs-pst** [comment]
@sylvioalves Can you re-trigger the CI ?

**Comment by sylvioalves** [comment]
@ndrs-pst I guess you need to rebase.

**Pull request merged**